baseapp/options_test.go

```diff
@@ -0,0 +1,18 @@
+package baseapp
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+)
+
+// TestSetAppVersion verifies that SetAppVersion does not panic if paramStore is nil.
+func TestSetAppVersion(t *testing.T) {
+	baseApp := NewBaseApp("test", nil, nil, nil)
+	baseApp.paramStore = nil // explicitly set to nil
+
+	require.NotPanics(t, func() {
+		baseApp.SetAppVersion(types.Context{}, uint64(1))
+	})
+}
```

Comment:
note this test failed before the change in this PR so this test should catch regressions if we attempt to merge in upstream changes from cosmos-sdk and accidentally revert the implementation change in options.go